processor/transaction/processor.go

```diff
@@ -20,22 +19,17 @@ const (
 
 func NewProcessor() pr.Processor {
 	return &processor{
-		payload: &payload{},
-		schema:  pr.CreateSchema(transactionSchema, processorName),
+		schema: pr.CreateSchema(transactionSchema, processorName),
 	}
 }
 
 type processor struct {
-	payload *payload
-	schema  *jsonschema.Schema
+	schema *jsonschema.Schema
 }
 
-func (p *processor) Transform() []beat.Event {
-	return p.payload.transform()
-}
-
-func (p *processor) Validate(reader io.Reader) error {
-	return pr.Validate(reader, p.schema, p.payload)
+func (p *processor) Validate(reader io.Reader) (pr.Payload, error) {
+	var payload Payload
+	return &payload, pr.Validate(reader, p.schema, &payload)
```

Comment:
what do you think about returning the validated `payload` from the `pr.Validate` method?